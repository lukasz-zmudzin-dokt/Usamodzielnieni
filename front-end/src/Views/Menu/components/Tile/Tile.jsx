import React from "react";
import { Link } from "react-router-dom";

const Tile = ({ title, showImage, imageUrl, color, destination }) => {
  const getClassNames = (name, position) =>
    `tile__${name} tile__${name}--${position} ${
      showImage[position] ? "tile__fragment--show" : ""
    }`;
  const positions = ["left", "top", "right"];

  return (
    <Link
      to={destination}
      className="tile__container"
      style={{ backgroundColor: color }}
    >
      <div className="tile">
        {positions.map((position) => (
          <div
            key={`bg_${position}`}
            style={{backgroundColor: color}}
            className={getClassNames("bg", position)}
          />
        ))}
        <div className="tile_content">
          {positions.map((position) => (
            <div
              key={`border_${position}`}
              className={getClassNames("border", position)}
            />
          ))}
          <div className="tile__image">
            <img alt="" src={imageUrl} />
          </div>
          <div className="tile_title">
            <span>{title}</span>
          </div>
        </div>
      </div>
    </Link>
  );
};

export default Tile;
