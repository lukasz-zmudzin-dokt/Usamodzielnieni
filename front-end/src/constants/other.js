const DEFAULT_INPUT = "-- Wybierz --";

export { DEFAULT_INPUT };
