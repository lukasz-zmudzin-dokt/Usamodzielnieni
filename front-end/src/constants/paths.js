const paths = {
    DASHBOARD: "/",
    CV_CREATOR: "/cvEditor",
    CV_EDITOR: "/cvEditor/:id",
    REGISTER: "/newAccount/",
    REGISTER_ADMIN: "/newAccount/:role(staff)",
    FOOTER: "/footer",
    LOGIN: "/login",
    USER: "/user",
    OFFER_FORM: "/offerForm/:id?",
    JOB_OFFERS: "/jobOffers",
    CV_APPROVAL: "/cvApproval",
    MY_OFFERS: "/myOffers",
    CONTACT_PAGE: "/contact",
    BLOG_POST: "/blog/blogpost/:id",
    BLOG_PAGE: "/blog",
    BLOG_FORM: "/blog/newPost",
    BLOG_EDIT: "/blog/newPost/:id",
    JOB_OFFER_DETAILS: "/jobOffers/:id",
    MY_CVS: "/myCVs",
    USER_APPROVAL: "/userApproval",
    OFFER_APPROVAL: "/offerApproval"
};
export {paths};