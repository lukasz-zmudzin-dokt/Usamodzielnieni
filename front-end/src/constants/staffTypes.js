export const staffTypes = {
  VERIFICATION: "staff_verification",
  CV: "staff_cv",
  JOBS: "staff_jobs",
  BLOG_CREATOR: "staff_blog_creator",
  BLOG_MODERATOR: "staff_blog_moderator",
};
