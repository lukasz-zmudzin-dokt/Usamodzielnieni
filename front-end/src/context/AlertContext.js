import React, { useState, createContext } from "react";

export const AlertContext = createContext({
  open: false,
  message: "",
  setOpen: () => {},
});

export const AlertProvider = (props) => {
  const [open, setOpen] = useState(false);
  const [message, setMessage] = useState("");
  const [variant, setVariant] = useState("danger") // success
  const data = {
    open,
    message,
    variant,
    changeVisibility: (newOpen) => setOpen(newOpen),
    changeMessage: (newMessage,newVariant = variant) => {
      setOpen(false);
      setMessage(newMessage);
      setVariant(newVariant);
    },
    showAlert:  (newMessage,newVariant) => {
      data.changeMessage(newMessage,newVariant);
      data.changeVisibility(true);
    }
  };
  return <AlertContext.Provider value={data} {...props} />;
};
