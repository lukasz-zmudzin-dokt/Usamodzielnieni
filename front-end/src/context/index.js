export { UserContext, UserProvider } from "context/UserContext";
export { AlertContext, AlertProvider } from "context/AlertContext";
export { ChatContext, ChatProvider } from "context/ChatContext";
export {
  NotificationsContext,
  NotificationsProvider,
} from "context/NotificationsContext";
