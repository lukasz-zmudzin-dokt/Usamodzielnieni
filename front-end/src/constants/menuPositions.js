import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
