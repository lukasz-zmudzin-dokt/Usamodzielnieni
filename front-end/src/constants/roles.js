import { staffTypes } from "./staffTypes";

export const adminGroup = [
  {
    name: staffTypes.VERIFICATION,
    placeholder: "Weryfikacja użytkowników",
  },
  {
    name: staffTypes.CV,
    placeholder: "Weryfikacja CV",
  },
  {
    name: staffTypes.JOBS,
    placeholder: "Weryfikacja ofert pracy",
  },
  {
    name: staffTypes.BLOG_CREATOR,
    placeholder: "Kreator postów na blogu",
  },
  {
    name: staffTypes.BLOG_MODERATOR,
    placeholder: "Moderator bloga / designer",
  },
  {
    name: staffTypes.GUEST,
    placeholder: "Konto gościa",
  },
  {
    name: staffTypes.CHAT,
    placeholder: "Dostęp do czatu",
  },
];

export const commonGroup = [
  {
    name: "Podopiecznym",
  },
  {
    name: "Pracodawcą",
  },
];
