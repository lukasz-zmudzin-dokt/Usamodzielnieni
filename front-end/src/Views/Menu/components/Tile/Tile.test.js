import React from "react";
import { render, wait, fireEvent } from "@testing-library/react";
import Tile from "./Tile";
import { MemoryRouter } from "react-router-dom";
import {AlertContext} from "context/AlertContext";
import {staffTypes} from "constants/staffTypes";

describe("Tile", () => {
  let props, appendTile, cutTile, apiFail, alertC;

  beforeAll(() => {
    global.fetch = jest.fn().mockImplementation(() => {
      return new Promise((resolve) => {
        if (apiFail) {
          resolve({status: 500, json: () => Promise.resolve({e: "e"})})
        } else {
          resolve({status: 200})
        }
      })
    })
  });

  beforeEach(() => {
    appendTile = jest.fn();
    cutTile = jest.fn();
    props = {
      id: 123,
      title: "Tytuł kafelka",
      showImage: { left: true, top: true, right: true },
      imageUrl: "/foo_bar.jpg",
      color: "#fff",
      destination: "/blog/abc123",
      user: {
        data: {
          group_type: [staffTypes.BLOG_MODERATOR]
        }
      }
    };
    alertC = {
      showAlert: jest.fn()
    }
  });

  it("should render without crashing", () => {
    const { container } = render(
        <AlertContext.Provider value={alertC}>
          <MemoryRouter>
            <Tile {...props} />
          </MemoryRouter>
        </AlertContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render without crashing when showImage object is empty", () => {
    props.showImage = {};
    const { container } = render(
        <AlertContext.Provider value={alertC}>
          <MemoryRouter>
            <Tile {...props} />
          </MemoryRouter>
        </AlertContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render edition modal', () => {
    const { getByRole, getByAltText } = render(
        <AlertContext.Provider value={alertC}>
          <MemoryRouter>
            <Tile {...props} />
          </MemoryRouter>
        </AlertContext.Provider>
    );

    fireEvent.click(getByAltText("Edytuj"));

    expect(getByRole("dialog")).toBeInTheDocument();
  });

  it('should delete tile', async () => {
    const { getByAltText, getByText } = render(
        <AlertContext.Provider value={alertC}>
          <MemoryRouter>
            <Tile {...props} cutTile={cutTile} />
          </MemoryRouter>
        </AlertContext.Provider>
    );

    fireEvent.click(getByAltText("Usuń"));
    fireEvent.click(getByText("Usuń ✗"));

    await wait(() => expect(fetch).toHaveBeenCalled());
    expect(alertC.showAlert).toHaveBeenCalledWith(
        "Kafelek usunięty pomyślnie.", "success"
    );

    expect(cutTile).toHaveBeenCalledWith(123);
  });

  it('should render preview of tile', () => {
    const { queryByAltText } = render(
        <AlertContext.Provider value={alertC}>
          <MemoryRouter>
            <Tile {...props} previewOnly/>
          </MemoryRouter>
        </AlertContext.Provider>
    );

    expect(queryByAltText("Edytuj")).not.toBeInTheDocument();
    expect(queryByAltText("Usuń")).not.toBeInTheDocument();
  })
});
