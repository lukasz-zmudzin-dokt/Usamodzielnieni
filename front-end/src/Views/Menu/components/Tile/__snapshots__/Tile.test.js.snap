// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile should render without crashing 1`] = `
<div>
  <div
    class="tile__container"
    style="background-color: rgb(255, 255, 255);"
  >
    <a
      href="/blog/abc123"
    >
      <div
        class="tile"
      >
        <div
          class="tile__bg tile__bg--left tile__fragment--show"
          style="background-color: rgb(255, 255, 255);"
        />
        <div
          class="tile__bg tile__bg--top tile__fragment--show"
          style="background-color: rgb(255, 255, 255);"
        />
        <div
          class="tile__bg tile__bg--right tile__fragment--show"
          style="background-color: rgb(255, 255, 255);"
        />
        <div
          class="tile_content"
        >
          <div
            class="tile__border tile__border--left tile__fragment--show"
          />
          <div
            class="tile__border tile__border--top tile__fragment--show"
          />
          <div
            class="tile__border tile__border--right tile__fragment--show"
          />
          <div
            class="tile__image"
          >
            <img
              alt=""
              src="/foo_bar.jpg"
            />
          </div>
          <div
            class="tile_title"
          >
            <span>
              Zaloguj się i Tytuł kafelka
            </span>
          </div>
        </div>
      </div>
    </a>
    <div
      class="tile__button__container"
    >
      <button
        class="tile__button tile__button__edit btn btn-primary btn-sm"
        type="button"
      >
        <img
          alt="Edytuj"
          class="tile__button__edit__icon"
          src="/editicon.svg"
        />
      </button>
      <button
        class="tile__button tile__button__delete btn btn-danger btn-sm"
        type="button"
      >
        <img
          alt="Usuń"
          class="tile__button__delete__icon"
          src="/deleteicon.svg"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Tile should render without crashing when showImage object is empty 1`] = `
<div>
  <div
    class="tile__container"
    style="background-color: rgb(255, 255, 255);"
  >
    <a
      href="/blog/abc123"
    >
      <div
        class="tile"
      >
        <div
          class="tile__bg tile__bg--left "
          style="background-color: rgb(255, 255, 255);"
        />
        <div
          class="tile__bg tile__bg--top "
          style="background-color: rgb(255, 255, 255);"
        />
        <div
          class="tile__bg tile__bg--right "
          style="background-color: rgb(255, 255, 255);"
        />
        <div
          class="tile_content"
        >
          <div
            class="tile__border tile__border--left "
          />
          <div
            class="tile__border tile__border--top "
          />
          <div
            class="tile__border tile__border--right "
          />
          <div
            class="tile__image"
          >
            <img
              alt=""
              src="/foo_bar.jpg"
            />
          </div>
          <div
            class="tile_title"
          >
            <span>
              Zaloguj się i Tytuł kafelka
            </span>
          </div>
        </div>
      </div>
    </a>
    <div
      class="tile__button__container"
    >
      <button
        class="tile__button tile__button__edit btn btn-primary btn-sm"
        type="button"
      >
        <img
          alt="Edytuj"
          class="tile__button__edit__icon"
          src="/editicon.svg"
        />
      </button>
      <button
        class="tile__button tile__button__delete btn btn-danger btn-sm"
        type="button"
      >
        <img
          alt="Usuń"
          class="tile__button__delete__icon"
          src="/deleteicon.svg"
        />
      </button>
    </div>
  </div>
</div>
`;
